.github/workflows/build.yml

```diff
@@ -93,9 +95,32 @@ jobs:
           name: ${{ steps.setup_precompiled.outputs.artifact_name }}
           path: ..
 
+  lint:
+    name: lint mathlib
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      # number of commits to check for olean cache
+      GIT_HISTORY_DEPTH: 20
+    steps:
+      - name: retrieve build
+        uses: actions/download-artifact@v1
+        with:
+          name: ${{ needs.build.outputs.artifact_name }}
+          path: ..
+
+      - name: install elan
+        run: |
+          set -o pipefail
+          curl https://raw.githubusercontent.com/Kha/elan/master/elan-init.sh -sSf | sh -s -- --default-toolchain none -y
+          ~/.elan/bin/lean --version
+          echo "::add-path::$HOME/.elan/bin"
+          echo "::set-env name=short_lean_version::$(~/.elan/bin/lean --run scripts/lean_version.lean)"
+
       - name: lint
         if: success() || steps.build.outputs.status == 'built'
```

Comment:
```suggestion
```
I think this can be deleted now? This was supposed to make it so that the `lint` and `test` steps would run only if the `build` step completed successfully, and `test` would still run even if `lint` failed.